src/client/session/mod.rs

```diff
@@ -37,6 +37,57 @@ lazy_static! {
 ///
 /// `ClientSession` instances are not thread safe or fork safe. They can only be used by one thread
 /// or process at a time.
+///
+/// ## Transactions
+/// Replica set transactions are supported on MongoDB 4.0+. Transactions are associated with a
+/// `ClientSession`. To begin a transaction, call [ClientSession::start_transaction] on a
+/// `ClientSession`. The `ClientSession` must be passed to operations to be executed within the
+/// transaction.
+///
+/// ```rust
+/// # use mongodb::{
+/// #     bson::doc,
+/// #     error::{Result, TRANSIENT_TRANSACTION_ERROR, UNKNOWN_TRANSACTION_COMMIT_RESULT},
+/// #     Client,
+/// #     ClientSession,
+/// #     Collection,
+/// # };
+/// # async fn do_stuff() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let coll = client.database("foo").collection("bar");
+/// # let mut session = client.start_session(None).await?;
+/// session.start_transaction(None).await?;
```

Comment:
I think it would be good to demonstrate best practices here and specify a majority read / write concern in the transaction options.